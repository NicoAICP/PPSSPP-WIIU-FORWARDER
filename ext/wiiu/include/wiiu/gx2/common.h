#pragma once

#define GX2_SCAN_BUFFER_ALIGNMENT      0x1000
#define GX2_SHADER_ALIGNMENT           0x100
#define GX2_CONTEXT_STATE_ALIGNMENT    0x100
#define GX2_UNIFORM_BLOCK_ALIGNMENT    0x100
#define GX2_DISPLAY_LIST_ALIGNMENT     0x20
#define GX2_VERTEX_BUFFER_ALIGNMENT    0x40
#define GX2_INDEX_BUFFER_ALIGNMENT     0x20

#define GX2_ENABLE                     TRUE
#define GX2_DISABLE                    FALSE

#define GX2_TRUE                       TRUE
#define GX2_FALSE                      FALSE

#ifndef GX2_COMP_SEL
#define GX2_COMP_SEL(c0, c1, c2, c3) (((c0) << 24) | ((c1) << 16) | ((c2) << 8) | (c3))
#ifdef __cplusplus
#include <functional>
#endif
#include <stdio.h>

#define _x 0
#define _y 1
#define _z 2
#define _w 3
#define _r 0
#define _g 1
#define _b 2
#define _a 3
#define _0 4
#define _1 5
#endif


#define GX2_DISABLE_WRAPS
