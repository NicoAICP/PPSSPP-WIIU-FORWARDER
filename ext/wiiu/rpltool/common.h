#ifndef COMMON_H
#define COMMON_H

#include <ctype.h>
#include <stdint.h>
#include <stdio.h>

typedef int8_t s8;
typedef int16_t s16;
typedef int32_t s32;

typedef uint8_t u8;
typedef uint16_t u16;
typedef uint32_t u32;

#define GET_ENUM_NAME(prefix, name, val) \
   case prefix##name: \
      return #name;

#define EMIT_ENUM_TO_STR_FUNC(name, cases) \
   static inline const char *name##_to_str(name type) { \
      switch (type) { cases } \
      return "unknown"; \
   }

#define GET_FLAG_NAME(prefix, name, val) \
   case prefix##name: \
      pos += snprintf(buffer + pos, sizeof(buffer) - pos, "%s%s", #name, seperator); \
      flags &= ~(1u << i); \
      break;

#define EMIT_FLAGS_TO_STR_FUNC(name, cases) \
   static inline const char *name##_to_str(int flags, const char *seperator) { \
      int i; \
      static char buffer[32 * 32]; \
      int pos = 0; \
      for (i = 0; i < 32; i++) \
         if ((1u << i) & flags) \
            switch (1u << i) { cases } \
      if (flags) \
         snprintf(buffer + pos, sizeof(buffer) - pos, "0x%08X", flags); \
      else if (pos) \
         buffer[pos - strlen(seperator)] = '\0'; \
      else \
         return "NULL"; \
      return buffer; \
   }

#define MAKE_MAGIC(a, b, c, d) (((a) << 24) | ((b) << 16) | ((c) << 8) | ((d) << 0))

#define log_var(X) printf("%-35s: 0x%0*X (%u)\n", #X, sizeof(X) * 2, (u32)(X), (u32)(X))
#define log_svar(X) printf("%-35s: 0x%0*X (%i)\n", #X, sizeof(X) * 2, (u32)(X), (s32)(X))
#define log_array(X) debug_array(X, sizeof(X), #X)
#define log_enum(type, X) printf("%-35s: 0x%X (%s)\n", #X, (X), type##_to_str(X))
#define log_flags(type, X) printf("%-35s: 0x%08X (%s)\n", #X, (X), type##_to_str(X, " | "))
#define log_magic(X) \
   printf("%-35s: '%c''%c''%c''%c' (0x%08X)\n", #X, (u32)(X) >> 24, (u32)(X) >> 16, (u32)(X) >> 8, (u32)(X), (u32)(X))
#define log_carray(X) debug_char_array(X, sizeof(X), #X)
#define log_str(X) printf("%-35s: \"%s\"\n", #X, (char *)(X))
#define log_off_str(X, data) printf("%-35s: 0x%08X \"%s\"\n", #X, (X), (char *)data + (X))
#define log_off_strs(X, data) \
   do { \
      printf("%-35s: 0x%08X ", #X, (X)); \
      const char *ptr = (char *)data + (X); \
      while (*ptr) { \
         ptr += printf("\"%s\" ", ptr) - 2; \
      } \
      printf("\n"); \
   } while (0)

static inline void dump_array(uint32_t *src, int size);
static inline void debug_array(uint32_t *src, int size, const char *name) {
   printf("%-35s:", name);
   dump_array(src, size);
}
static inline void debug_char_array(uint8_t *src, int size, const char *name) {
   int i;
   printf("%-35s: ", name);
   //   for (i = 0; i < size; i++)
   //      printf("%-2c ", isalnum(src[i])?src[i]: '?');
   //   printf("\n%-35s: ", "");
   for (i = 0; i < size; i++)
      printf("%02X ", src[i]);
   printf("\n");
}

static inline void dump_array(uint32_t *src, int size) {
   int i;
   size = (size / 4) - 1;
   printf("\n{");
   for (i = 0; i < size; i++) {
      if (!(i & 3))
         printf("\n   ");

      printf("0x%08X, ", src[i]);
   }

   if (!(i & 3))
      printf("\n   ");

   printf("0x%08X\n}\n", src[i]);
}

static inline u32 align_up(u32 x, u32 align) { return (x + (align - 1)) & ~(align - 1); }
static inline u32 align_down(u32 x, u32 align) { return x & ~(align - 1); }

#endif // COMMON_H
