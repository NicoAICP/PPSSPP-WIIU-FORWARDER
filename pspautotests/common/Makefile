TARGET_LIB = libcommon.a
EXTRA_TARGETS = libcommon_kernel.a
OBJS = common.o vram.o sysmem-imports.o
KERNEL_OBJS = common_kernel.o vram.o sysmem-imports.o

INCDIR = 
CFLAGS = -O0 -G 0 -g -Wall
CXXFLAGS = $(CFLAGS) -fno-exceptions -fno-rtti
ASFLAGS = $(CFLAGS)

LIBDIR =
LDFLAGS = -G0

PSPSDK=$(shell psp-config --pspsdk-path)
include $(PSPSDK)/lib/build.mak

%.o: %.S
	$(AS) $(ASFLAGS) -c -o $@ $<

common_kernel.o: common.c
	$(CC) $(CFLAGS) -DCOMMON_KERNEL=1 -c -o $@ $<

libcommon_kernel.a: $(KERNEL_OBJS)
	$(AR) cru $@ $(KERNEL_OBJS)
	$(RANLIB) $@
