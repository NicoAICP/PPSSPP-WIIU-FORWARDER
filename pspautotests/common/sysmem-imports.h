extern int sceKernelSetCompiledSdkVersion(int ver);
extern int sceKernelSetCompiledSdkVersion370(int ver);
extern int sceKernelSetCompiledSdkVersion380_390(int ver);
extern int sceKernelSetCompiledSdkVersion395(int ver);
extern int sceKernelSetCompiledSdkVersion401_402(int ver);
extern int sceKernelSetCompiledSdkVersion500_505(int ver);
extern int sceKernelSetCompiledSdkVersion507(int ver);
extern int sceKernelSetCompiledSdkVersion600_602(int ver);
extern int sceKernelSetCompiledSdkVersion603_605(int ver);
extern int sceKernelSetCompiledSdkVersion606(int ver);
extern int sceKernelGetCompiledSdkVersion();