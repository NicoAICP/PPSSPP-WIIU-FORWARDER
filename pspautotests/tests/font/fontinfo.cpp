#include <common.h>
#include <pspmodulemgr.h>
#include <pspiofilemgr.h>
#include <malloc.h>
extern "C" {
#include "libfont.h"
}

static FontLibraryHandle fontLib;
static FontHandle font;
static FontHandle fontClosed;
static int g_allocated = 0;

static void *allocQuiet(void *arg, u32 size) {
	g_allocated += size;
	void *p = malloc(size);
	//checkpoint("%08x = alloc(0x%x)", p, size);
	return p;
}

static void freeQuiet(void *arg, void *p) {
	//checkpoint("free(%08x)", p);
	free(p);
}

bool loadFontModule() {
	checkpointNext("Init");
	if (RUNNING_ON_EMULATOR) {
		return true;
	}
	SceUID fontModule = sceKernelLoadModule("libfont.prx", 0, NULL);
	if (fontModule <= 0) {
		printf("TEST ERROR: Unable to load libfont.prx\n");
		return false;
	}

	int status = -1;
	int result = sceKernelStartModule(fontModule, 0, NULL, &status, NULL);
	if (result != fontModule || status != 0) {
		printf("TEST ERROR: libfont.prx startup failed (%08x, %08x)\n", result, status);
		return false;
	}
	return true;
}

bool loadFontHandles() {
	FontNewLibParams libParams;
	memset(&libParams, 0, sizeof(libParams));
	libParams.allocFuncAddr = allocQuiet;
	libParams.freeFuncAddr = freeQuiet;
	libParams.numFonts = 4;

	uint error = -1;
	fontLib = sceFontNewLib(&libParams, &error);

	font = sceFontOpenUserFile(fontLib, "ltn0.pgf", 1, &error);
	if (font <= 0) {
		printf("TEST ERROR: Unable to load ltn0.pgf\n");
		return false;
	}

	fontClosed = sceFontOpenUserFile(fontLib, "ltn0.pgf", 1, &error);
	sceFontClose(fontClosed);

	return true;
}

void schedfStyle(FontStyle *style) {
	schedf("Font Style: h=%f (%f), v=%f (%f), w=%f", style->fontH, style->fontHRes, style->fontV, style->fontVRes, style->fontWeight);
	schedf(", family=%d, style=%d / %d", style->fontFamily, style->fontStyle, style->fontStyleSub);
	schedf(", lang=%d, reg=%d, country=%d", style->fontLanguage, style->fontRegion, style->fontCountry);
	schedf(", name=%s, file=%s", style->fontName, style->fontFileName);
	schedf(", attr=%08x, exp=%d\n", style->fontAttributes, style->fontExpire);
}

void schedfInfo(FontInfo *info) {
	schedf("Font Info: maxGlyphWidthI: %d", info->maxGlyphWidthI);
	schedf(", maxGlyphHeightI: %d", info->maxGlyphHeightI);
	schedf(", maxGlyphAscenderI: %d", info->maxGlyphAscenderI);
	schedf(", maxGlyphDescenderI: %d", info->maxGlyphDescenderI);
	schedf(", maxGlyphLeftXI: %d", info->maxGlyphLeftXI);
	schedf(", maxGlyphBaseYI: %d", info->maxGlyphBaseYI);
	schedf(", minGlyphCenterXI: %d", info->minGlyphCenterXI);
	schedf(", maxGlyphTopYI: %d", info->maxGlyphTopYI);
	schedf(", maxGlyphAdvanceXI: %d", info->maxGlyphAdvanceXI);
	schedf(", maxGlyphAdvanceYI: %d", info->maxGlyphAdvanceYI);

	schedf(", maxGlyphWidthF: %f", info->maxGlyphWidthF);
	schedf(", maxGlyphHeightF: %f", info->maxGlyphHeightF);
	schedf(", maxGlyphAscenderF: %f", info->maxGlyphAscenderF);
	schedf(", maxGlyphDescenderF: %f", info->maxGlyphDescenderF);
	schedf(", maxGlyphLeftXF: %f", info->maxGlyphLeftXF);
	schedf(", maxGlyphBaseYF: %f", info->maxGlyphBaseYF);
	schedf(", minGlyphCenterXF: %f", info->minGlyphCenterXF);
	schedf(", maxGlyphTopYF: %f", info->maxGlyphTopYF);
	schedf(", maxGlyphAdvanceXF: %f", info->maxGlyphAdvanceXF);
	schedf(", maxGlyphAdvanceYF: %f", info->maxGlyphAdvanceYF);
		
	schedf(", maxGlyphWidth: %d, maxGlyphHeight: %d", info->maxGlyphWidth, info->maxGlyphHeight);
	schedf(", charMapLength: %d, shadowMapLength: %d", info->charMapLength, info->shadowMapLength);

	schedf(", pad0=%02x, pad1=%02x, pad2=%02x", info->pad[0], info->pad[1], info->pad[2]);

	schedf(", BPP: %d\n", info->BPP);
	schedfStyle(&info->fontStyle);
}

void testFontInfo(const char *title, FontHandle f, FontInfo *fontInfo) {
	if (fontInfo) {
		memset(fontInfo, 0xdd, sizeof(FontInfo));
	}
	int result = sceFontGetFontInfo(f, fontInfo);
	if (result == 0) {
		checkpoint(NULL);
		schedf("%s: OK ", title);
		schedfInfo(fontInfo);
	} else {
		checkpoint("%s: Failed (%08x)", title, result);
	}
}

void testFont() {
	FontInfo fontInfo;

	checkpointNext("Fonts:");
	testFontInfo("  Normal", font, &fontInfo);
	testFontInfo("  NULL", 0, &fontInfo);
	testFontInfo("  Closed", fontClosed, &fontInfo);

	checkpointNext("Info:");
	testFontInfo("  Missing", font, NULL);

	u32 data[69];
	memset(data, 0xdd, sizeof(data));
	sceFontGetFontInfo(font, (FontInfo *)data);
	checkpoint("  Data size: %08x %08x %08x %08x", data[65], data[66], data[67], data[68]);
}

extern "C" int main(int argc, char *argv[]) {
	if (!loadFontModule()) {
		return 1;
	}
	if (!loadFontHandles()) {
		return 1;
	}

	testFont();

	sceFontClose(font);
	sceFontDoneLib(fontLib);

	return 0;
}