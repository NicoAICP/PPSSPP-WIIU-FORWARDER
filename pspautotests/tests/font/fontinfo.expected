[x] Init

[x] Fonts:
[x]   Normal: OK Font Info: maxGlyphWidthI: 1455, maxGlyphHeightI: 1222, maxGlyphAscenderI: 863, maxGlyphDescenderI: -196, maxGlyphLeftXI: 195, maxGlyphBaseYI: 1202, minGlyphCenterXI: -727, maxGlyphTopYI: 18, maxGlyphAdvanceXI: 1585, maxGlyphAdvanceYI: 1094, maxGlyphWidthF: 22.734375, maxGlyphHeightF: 19.093750, maxGlyphAscenderF: 13.484375, maxGlyphDescenderF: -3.062500, maxGlyphLeftXF: 3.046875, maxGlyphBaseYF: 18.781250, minGlyphCenterXF: -11.359375, maxGlyphTopYF: 0.281250, maxGlyphAdvanceXF: 24.765625, maxGlyphAdvanceYF: 17.093750, maxGlyphWidth: 23, maxGlyphHeight: 20, charMapLength: 324, shadowMapLength: 0, pad0=dd, pad1=dd, pad2=dd, BPP: 4
Font Style: h=10.125000 (128.000000), v=10.125000 (128.000000), w=0.000000, family=0, style=0 / 0, lang=0, reg=0, country=0, name=, file=, attr=00000000, exp=0
[x]   NULL: Failed (80460003)
[x]   Closed: OK Font Info: maxGlyphWidthI: 1455, maxGlyphHeightI: 1222, maxGlyphAscenderI: 863, maxGlyphDescenderI: -196, maxGlyphLeftXI: 195, maxGlyphBaseYI: 1202, minGlyphCenterXI: -727, maxGlyphTopYI: 18, maxGlyphAdvanceXI: 1585, maxGlyphAdvanceYI: 1094, maxGlyphWidthF: 22.734375, maxGlyphHeightF: 19.093750, maxGlyphAscenderF: 13.484375, maxGlyphDescenderF: -3.062500, maxGlyphLeftXF: 3.046875, maxGlyphBaseYF: 18.781250, minGlyphCenterXF: -11.359375, maxGlyphTopYF: 0.281250, maxGlyphAdvanceXF: 24.765625, maxGlyphAdvanceYF: 17.093750, maxGlyphWidth: 23, maxGlyphHeight: 20, charMapLength: 324, shadowMapLength: 0, pad0=dd, pad1=dd, pad2=dd, BPP: 4
Font Style: h=10.125000 (128.000000), v=10.125000 (128.000000), w=0.000000, family=0, style=0 / 0, lang=0, reg=0, country=0, name=, file=, attr=00000000, exp=0

[x] Info:
[x]   Missing: Failed (80460003)
[x]   Data size: dddddd04 dddddddd dddddddd dddddddd
